node/src/actors/chain_manager/mod.rs

```diff
@@ -969,18 +972,64 @@ impl ChainManager {
             }
         })
         .map_err(|e, _, _| log::error!("Superblock building failed: {:?}", e))
-        .and_then(move |(block_headers, last_hash), act, _ctx| {
-            let ars_members = &act.chain_state.last_ars;
-            let ars_ordered_keys = &act.chain_state.last_ars_ordered_keys;
-
-            let superblock = act.chain_state.superblock_state.build_superblock(
-                &block_headers,
-                ars_members,
-                ars_ordered_keys,
-                superblock_index,
-                last_hash,
-            );
-            actix::fut::ok(superblock)
+        .and_then(move |(block_headers, last_hash), act, ctx| {
+            let consensus = if act.sm_state == StateMachine::Synced {
```

Comment:
This should be changed to `Synced || AlmostSynced`

In `Synchronizing` we should assume the superblock is valid, but in `WaitingConsensus` we should not create superblocks.